.github/workflows/deploy.yml

```diff
@@ -0,0 +1,30 @@
+name: Deploy
+
+on:
+  push:
+    tags: v[0-9]+.[0-9]+.[0-9]+
```

Comment:
Have you configured a tag protection rule for this pattern? https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-tag-protection-rules

Reply:
I have a rule for `v*`, but regardless of this the deploy won't happen without an approval of the `Deploy` environment. It's not super obvious that that's been configured in the GitHub Action config itself, we'll have to test it out once we merge this.

Reply:
Ah yes, I overlooked `environment: Deploy`. I haven't used environments yet but am eager to see how it works.